adafruit_emc2101/emc2101_fanspeed.py

```diff
@@ -0,0 +1,202 @@
+
+        emc2101.lut[50] = 62
+
+    Set two values up and write to the chip once.
+
+    .. code-block:: python
+
+        with emc2101.lut as lut:
+            lut[20] = 0
+            lut[40] = 10
+
+    Delete an unneeded temperature point: (Updates the chip immediately).
+
+    .. code-block:: python
+
+        emc2101.lut[20] = None
+
+    Read a dict of the currently set values:
+
+    .. code-block:: python
+
+        values = emc2101.lut.lookup_table
+        # returns:
+        # { 40: 10, 50: 62 }
+
+    Delete all current LUT values:
+
+    .. code-block:: python
+
+        for temp in emc2101.lut.lookup_table:
+            emc2101.lut[temp] = None
+    """
+
+    # 8 (Temperature, Speed) pairs in increasing order
+    _fan_lut = StructArray(emc2101_regs.LUT_BASE, "<B", 16)
+    _defer_update = 0
+
+    def __init__(self, fan_obj):
+        self._defer_update = 0
+        self.emc_fan = fan_obj
+        self.lut_values = {}
+        self.i2c_device = fan_obj.i2c_device
+
+    def __enter__(self):
+        """'with' wrapper: defer lut update until end of 'with' so
+        update_lut work can be done just once at the end of setting the LUT.
+
+        """
+        # Use increment/decrement so nested with's are dealt with properly.
+        self._defer_update += 1
+        return self
+
+    # 'with' wrapper
+    def __exit__(self, typ, val, tbk):
+        """'with' wrapper: defer lut update until end of 'with' so
+        update_lut work can be done just once at the end of setting the LUT.
+        """
+        self._defer_update -= 1
+        if self._defer_update == 0:
+            self._update_lut()
+
+    def __getitem__(self, index):
+        if not isinstance(index, int):
+            raise IndexError
+        if not index in self.lut_values:
+            raise IndexError
+        return self.lut_values[index]
+
+    def __setitem__(self, index, value):
+        if not isinstance(index, int):
+            raise IndexError
+        if value is None:
+            # Assign None to remove this entry
+            del self.lut_values[index]
+        elif value > 100.0 or value < 0:
+            # Range check
+            raise ValueError("LUT values must be a fan speed from 0-100%")
+        else:
+            self.lut_values[index] = value
+        if self._defer_update == 0:
+            self._update_lut()
+
+    def __repr__(self):
+        """return the official string representation of the LUT"""
+        # pylint: disable=consider-using-f-string
+        return "FanSpeedLUT {:x}".format(id(self))
+
+    def __str__(self):
+        """return the official string representation of the LUT"""
+        value_strs = []
+        lut_keys = tuple(sorted(self.lut_values.keys()))
+        for temp in lut_keys:
+            fan_drive = self.lut_values[temp]
+            # pylint: disable=consider-using-f-string
+            value_strs.append("%d deg C => %.1f%% duty cycle" % (temp, fan_drive))
+
+        return "\n".join(value_strs)
```

Comment:
I would be in favor of removing these.  I think they're nice to haves but I already worry about the size.  If someone really needs a string representation of some kind, I think having them build it in user code is worth the save here.